src/fondant/dataset/dataset.py

```diff
@@ -512,11 +549,7 @@ def read(
         Returns:
             An intermediate dataset.
         """
-        if self._graph:
-            msg = "For now, at most one read component can be applied per pipeline."
-            raise InvalidPipelineDefinition(
-                msg,
-            )
+        # TODO: add method call to retrieve workspace context, and make passing workspace optional
```

Comment:
Every operation method (read, apply, write) takes the workspace as an argument. 
@GeorgesLorre I guess we can use the workspace initialised in the `cli` to pass it somehow to these methods. Maybe as part of arguments? I would propose to clean this up as part of #887.

Reply:
This is indeed not ideal. Ideally we inject these values at compile time. That way we can keep the datasets separate from the workspace.